lib/fetchPosts.tsx

```diff
@@ -1,48 +1,47 @@
-import { FeedType } from '@/providers/FeedProvider';
 import { SupabasePost } from '@/types/SupabasePost';
 import { SupabaseClient } from '@supabase/supabase-js';
+import getFollowing from './neynar/getFollowing';
+import { fetchPostsLimit } from './consts';
+import { FeedType } from '@/types/Feed';
 
 const fetchPosts = async (
   supabaseClient: SupabaseClient,
   filter: any,
   feedType: string,
   start: number,
+  fid?: number,
 ) => {
+  let query;
   if (feedType === FeedType.Recent) {
-    const query = supabaseClient.from('posts').select('*').not('likes', 'is', null);
-
-    if (filter?.platform) {
-      query.eq('platform', filter.platform);
-    }
-
-    if (filter?.channel) {
-      query.eq('channelId', filter.channel);
-    }
-
+    query = supabaseClient.from('posts').select('*').not('likes', 'is', null);
     query.order('created_at', { ascending: false });
-    query.range(start, start + 20);
-
-    const { data: posts } = await query.returns<SupabasePost[]>();
-
-    return {
-      posts,
-    };
-  } else {
-    const query = supabaseClient.from('trending_posts').select('*');
+  } else if (feedType === FeedType.Trending) {
```

Comment:
nit: moving forward can you please consider `return` for cleaner code than `else if`?

learn more: https://stackoverflow.com/questions/9267643/if-return-vs-if-else-efficiency